sql: populate `pg_catalog.pg_matviews`

rohany opened the pull request:
Fixes #53492.

Release justification: low-risk updates to new functionality
Release note (sql change): Populate the catalog table
`pg_catalog.pg_matviews`.

rohany committed:
sql: populate `pg_catalog.pg_matviews`

Fixes #53492.

Release justification: low-risk updates to new functionality
Release note (sql change): Populate the catalog table
`pg_catalog.pg_matviews`.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53501)
<!-- Reviewable:end -->

solongordon reviewed (approved):
I'm curious about `hasindexes` but LGTM.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53501#-:-MFgmR-T7mqfeYDEeqMp:b-f78bk6)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @rohany)

___
*[pkg/sql/pg_catalog.go, line 1917 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/53501#-MFgklPL3n5_RLzhaMl9:-MFgklPL3n5_RLzhaMlA:b-sm1ec5) ([raw file](https://github.com/cockroachdb/cockroach/blob/b581e4f41b9581540cf54bdaab86daccc6496f2e/pkg/sql/pg_catalog.go#L1917)):*
> ```Go
> 				// we'll only print `SELECT b FROM foo` as the view definition here,
> 				// while postgres would more accurately print `SELECT b AS a FROM foo`.
> 				// TODO(a-robinson): Insert column aliases into view query once we
> ```

I know this was copy/pasted but it's kind of funny to insert a TODO for someone who doesn't work here anymore. :)
___
*[pkg/sql/logictest/testdata/logic_test/pg_catalog, line 324 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/53501#-MFglvp10e66_GSkaYky:-MFglvp10e66_GSkaYkz:bcdt0i4) ([raw file](https://github.com/cockroachdb/cockroach/blob/b581e4f41b9581540cf54bdaab86daccc6496f2e/pkg/sql/logictest/testdata/logic_test/pg_catalog#L324)):*
> ```
> ----
> schemaname  matviewname  matviewowner  tablespace  hasindexes  ispopulated  definition
> public      mv1          NULL          NULL        true        true         SELECT 1
> ```

Is this really supposed to have `hasindexes = true`? Is it because of the primary index? Is `hasindexes` ever false?


<!-- Sent from Reviewable.io -->

rohany commented:
> I know this was copy/pasted but it's kind of funny to insert a TODO for someone who doesn't work here anymore. :)

Didn't even notice that. Updated.

> Is this really supposed to have hasindexes = true? Is it because of the primary index? Is hasindexes ever false?

This is a good catch. It's supposed to be "has any secondary indexes". Updated.

rohany commented:
bors r+
